extensions/amp-accordion/1.0/accordion.js

```diff
@@ -88,19 +92,31 @@ function AccordionWithRef(
           expandSingleSection
         );
       });
-      return () => setExpandedMap((expandedMap) => omit(expandedMap, id));
+      eventMap.current[id] = {onExpand, onCollapse};
+      return () => {
+        setExpandedMap((expandedMap) => omit(expandedMap, id));
+        omit(/** @type {!Object} */ (eventMap.current), id);
+      };
     },
     [expandSingleSection]
   );
 
   const toggleExpanded = useCallback(
     (id) => {
+      const newValue = !expandedMap[id];
       setExpandedMap((expandedMap) => {
-        const newValue = !expandedMap[id];
         return setExpanded(id, newValue, expandedMap, expandSingleSection);
       });
+      const onExpand = eventMap.current[id] && eventMap.current[id].onExpand;
```

Comment:
Syntax sugar: this is a place where you could use something like `EMPTY_EVENT_MAP` like this:

```
const {onExpand, onCollapse} = eventMap.current[id] || EMPTY_EVENT_MAP;
if (onExpand) {...} else {onCollapse} {}
```

Reply:
Thanks, this is very helpful, although ultimately went with a single function to handle both `collapse` and `expand` so it looks slightly different.